lib/actions/field-trip.js

```diff
@@ -249,17 +406,355 @@ function planInbound (request) {
+
+  const res = await fetch(
+    `${datastoreUrl}/fieldtrip/getTrips?${qs.stringify(params)}`
+  )
+  const fieldTrips = await res.json()
+
+  // add passengers and converted tripId to trips
+  const trips = []
+  fieldTrips.forEach(fieldTrip => {
+    fieldTrip.groupItineraries.forEach(groupItinerary => {
+      groupItinerary.trips.forEach(gtfsTrip => {
+        // tripIds still stored as 'agencyAndId' in DB, so convert them to
+        // be compatible with OTP responses
+        gtfsTrip.tripId = gtfsTrip.agencyAndId.replace('_', ':')
+        // Add passengers to each trip from group itinerary
+        gtfsTrip.passengers = groupItinerary.passengers
+        trips.push(gtfsTrip)
+      })
+    })
+  })
+  await dispatch(receivedTravelDateTrips(trips))
+}
+
+/**
+ * Makes the needed requests to the OTP index API to obtain semantic trip hashes
+ * for any tripIds in the active itineraries that haven't already been obtained
+ * from the OTP index API.
+ */
+function getTripHashesFromActiveItineraries () {
+  return function (dispatch, getState) {
+    const state = getState()
+    const activeItineraries = getActiveItineraries(state)
+    const { tripHashLookup } = state.otp.callTaker.fieldTrip
+    const tripHashesToRequest = []
+    activeItineraries.forEach(itinerary => {
+      itinerary.legs.forEach(leg => {
+        if (leg.tripId && !tripHashLookup[leg.tripId]) {
+          tripHashesToRequest.push(leg.tripId)
+        }
+      })
+    })
+
+    const api = state.config.api
+    const baseUrl = `${api.host}${api.port ? ':' + api.port : ''}${api.path}`
+
+    return Promise.all(tripHashesToRequest.map(tripId => {
+      return fetch(`${baseUrl}/index/trips/${tripId}/semanticHash`)
+        .then(res => res.text())
+        .then(hash => dispatch(receiveTripHash({ hash, tripId })))
+    }))
+  }
+}
+
+/**
+ * Analyzes the current itineraries from each response in the active search to
+ * appropriately assign the field trip group to subgroups that use each
+ * itinerary according to the capacity available in each itinerary.
+ *
+ * @return {Object} result
+ * @return {Object} result.assignedItinerariesByResponse An Object organized by
+ *  response index and list of itineraries that can be used to update the state
+ *  with modified itinerary objects that can help display the field trip group
+ *  size assigned to each itinerary.
+ * @return {Object} result.remainingGroupSize The remaining about of people in
+ *  the field trip group that have yet to be assigned to an itinerary.
+ * @return {Object} result.tripsToBanInSubsequentSearches An array of strings
+ *  representing tripIds that should be added to the list of tripIds to ban when
+ *  making requests for additional itineraries from OTP.
+ */
+function checkValidityAndCapacity (state, request) {
+  const fieldTripModuleConfig = getModuleConfig(state, Modules.FIELD_TRIP)
+  const minimumAllowableRemainingCapacity =
+    fieldTripModuleConfig?.minimumAllowableRemainingCapacity || 10
```

Comment:
Perhaps we should just return a default field trip config from `getModuleConfig` if it does not exist, so that we can safely destructure (and avoid the optional chaining)?

Reply:
I'd prefer the `getModuleConfig` method to return something non-truthy if a module doesn't exist.